storage.go

```diff
@@ -3,34 +3,187 @@ package ftp
 import (
 	"context"
 	"io"
+	"path/filepath"
 
+	"github.com/beyondstorage/go-storage/v4/pkg/iowrap"
 	. "github.com/beyondstorage/go-storage/v4/types"
+	"github.com/jlaffaye/ftp"
+	mime "github.com/qingstor/go-mime"
 )
 
+type listDirInput struct {
+	rp  string
+	dir string
+
+	started           bool
+	continuationToken string
+	objChan           []*ftp.Entry
+	counter           int
+	buf               []byte
```

Comment:
I suggest three changes here:

1. Rename `objChan`, since it's not a `chan`, the name would lead misunderstanding. Maybe `objList`?
2. `counter` seems counted, but not used.
3. `buf` seems not used at all, what's the `buf` for?

Reply:
I feel a little confused with storage.list function. Maybe I need to refactor it later.

Reply:
Now I understand and I've fixed it.